giraffe/src/utils/mosaicTooltip.ts

```diff
@@ -64,10 +87,16 @@ export const getMosaicTooltipData = (
   const xMinCol = boxTable.getColumn(X_MIN, 'number')
   const xMaxCol = boxTable.getColumn(X_MAX, 'number')
   const valCol = boxTable.getColumn(FILL, 'string')
-  const yCol = boxTable.getColumn(SERIES, 'string')
+  const yCol = boxTable.getColumn(DISPLAY_NAME, 'string')
   const xFormatter = columnFormatter(xColKey)
   const yFormatter = columnFormatter(yColKey)
   const valFormatter = columnFormatter(FILL)
+
+  // Use reverse order because
+  // Mosaic graph contents are filled in from bottom to top
+  // Mosaic tooltip contents are filled in from top to bottom
+  const hoveredBoxRows = hoveredRowIndices.reverse()
+
```

Comment:
the code below is looping over hoveredBoxRows a number of separate times to get a different value. do you think we could do that loop once and store all the values?

`colors`, `xTooltipColumn.values`, `yTooltipColumn.values`, `durationTooltipColumn.values`, and `fillColumns.values`

Reply:
We could do that. From a performance standpoint though, i think they are the same. Looping five times through an array and doing a unique action per loop is the same as Looping through the array once but stopping at each element for 5 unique actions. None of the actions in the loops are repeated so i don't think there are any performance gains there.

Seeing `hoveredBoxRows.map()` called 5 different times is very eye catching for sure. And this conversation might be repeated in the future if we don't address it now.

Reply:
Reviewing this further, I don't see a clean way of refactoring this. We could put all the calls into an object, but we would still be making 5 separate calls. And now we just have a giant object with some properties with keys that have to be named something other than "values". I think this approach makes the code more verbose.

I like the .map calls in their current locations--right where they are needed.

Reply:
cool, fair enough. thank you for looking into it and getting an answer on this for the future.